pkg/vere/io/http.c

```diff
@@ -650,20 +650,38 @@ _http_req_dispatch(u3_hreq* req_u, u3_noun req)
 /* _http_cache_respond(): respond with a simple-payload:http
 */
 static void
-_http_cache_respond(u3_hreq* req_u, u3_noun nun) {
-  // XX check auth
-  u3_noun auth, response_header, data;
-  u3x_qual(u3k(u3t(u3t(nun))), &auth, 0, &response_header, &data);
-  u3_noun status, headers;
-  u3x_cell(response_header, &status, &headers);
+_http_cache_respond(u3_hreq* req_u, u3_noun nun, u3_noun req) {
+  h2o_req_t* rec_u = req_u->rec_u;
+  u3_httd* htd_u = req_u->hon_u->htp_u->htd_u;
 
-  req_u->sat_e = u3_rsat_plan;
-  _http_start_respond(req_u, u3k(status), u3k(headers), u3k(data), c3y);
+  if ( u3_none == nun ) {
+    h2o_send_error_500(rec_u, "Internal Server Error", "scry failed", 0);
+  }
+  else if ( (u3_none == u3t(nun)) || (u3_none == u3t(u3t(nun))) ) {
+    h2o_send_error_404(rec_u, "Not Found", "not found", 0);
+  }
+  else {
+    u3_noun auth, response_header, data;
+    u3x_qual(u3k(u3t(u3t(nun))), &auth, 0, &response_header, &data);
+    u3_noun status, headers;
+    u3x_cell(response_header, &status, &headers);
+
+    // check auth
+    c3_o aut_o = (c3n == auth) || _http_req_is_auth(&htd_u->fig_u, rec_u);
+    if ( c3n == aut_o ) {
+      h2o_send_error_403(rec_u, "Unauthorized", "unauthorized", 0);
+    }
+    else {
+      req_u->sat_e = u3_rsat_plan;
+      _http_start_respond(req_u, u3k(status), u3k(headers), u3k(data), c3y);
+    }
+  }
   u3z(nun);
 }
 
 typedef struct _cache_scry_cb_t {
   u3_hreq* req_u;
+  u3_noun  req;
```

Comment:
It looks like `req` is no longer used, can you remove it from here and `_http_cache_respond()`?